doc/source/ScriptPerformance/index.md

```diff
@@ -0,0 +1,17 @@
+# Script Performance #
+
+As with most VFX processes and software, Gaffer scripts – in both senses of your project files and the code that goes into them – are subject to performance considerations. If you're not careful, your graph can easily compute excessive recursion or unintendedly impactful processes, and the structure that brought about the drop in performance can be difficult to spot. Developing a basic understanding of how Gaffer evaluates graphs, and the complexity pitfalls users commonly encounter, can help you mitigate performance setbacks.
```

Comment:
> in both senses of your project files and the code that goes into them

Perhaps "your node graphs and the assets they process"? To me that would be a more specific description of the things the user is in change of, with the code more being a developer's problem.

> If you’re not careful, your graph can easily compute excessive recursion or unintendedly impactful processes, and the structure that brought about the drop in performance can be difficult to spot.

I know what you're getting at with "compute excessive recursion" and "unintendly impactful processes", but the phrases seem clumsy to me - almost pseudo-techy. Do we even need this sentence?

> can help you mitigate performance setbacks

can help you ensure optimum performance? :)

Reply:
> Perhaps "your node graphs and the assets they process"?

Since "script" can be misleading, I was (again) trying to distinguish between "Gaffer script" and what we know as "code and statements in plugs in the graph". My latest commit should make this clearer.